packages/config/src/config/config.utils.ts

```diff
@@ -55,13 +65,42 @@ export function isMultiCurrency<Symbols extends AssetSymbol = AssetSymbol>(
 export function getSymbol<Symbols extends AssetSymbol>(
   symbolOrAsset: Symbols | Asset<Symbols>,
 ): Symbols {
-  return typeof symbolOrAsset === 'string'
-    ? symbolOrAsset
-    : symbolOrAsset.originSymbol;
+  return isString(symbolOrAsset) ? symbolOrAsset : symbolOrAsset.originSymbol;
 }
 
 export function getChainKey<ChainKeys extends ChainKey>(
   keyOrChain: ChainKeys | Chain<ChainKeys>,
 ): ChainKeys {
-  return typeof keyOrChain === 'string' ? keyOrChain : keyOrChain.key;
+  return isString(keyOrChain) ? keyOrChain : keyOrChain.key;
+}
+
+export function getOverallWeight<ChainKeys extends ChainKey = ChainKey>(
+  chain: Chain<ChainKeys> | MoonChain,
+  txWeight: bigint,
+): bigint {
+  if (!chain.weights) {
+    throw new Error(`No weights found for chain: ${chain.key}`);
+  }
+
+  const {
+    descendOriginWeight,
+    withdrawAssetWeight,
+    buyExecutionWeight,
+    transactWeight,
+  } = chain.weights;
+
+  return (
+    descendOriginWeight +
+    withdrawAssetWeight +
+    buyExecutionWeight +
+    transactWeight +
+    txWeight
+  );
+}
+
+export function getOverallFee(
+  overallWeight: bigint,
+  unitsPerSecond: bigint,
+): bigint {
+  return (overallWeight * unitsPerSecond) / 1_000_000_000_000n;
```

Comment:
Maybe call this `weightPerSecond` so it is more clear?

Reply:
Ok, we can rename it.

Reply:
I mean only the `1_000_000_000_000n`